### ISSUE TITLE
Aerial Companion incompatible with Ventura 13.5.2

### ISSUE THREAD

**Issue by Gym-AW** [JohnCoates/Aerial#1320]
#### General troubleshooting tips

Before logging an issue please check that:
- [x] You have the latest version installed (There may be a beta version that fixes your issue), see here for the latest releases and bug fixes: https://github.com/JohnCoates/Aerial/releases
- [x] Your issue isn't already mentioned in our [issues](https://github.com/JohnCoates/Aerial/issues). You may find a workaround there or a similar request already made.
- [x] Your problem isn't mentioned in the [troubleshooting page](https://github.com/JohnCoates/Aerial/blob/master/Documentation/Troubleshooting.md).

If none of this fixes your issue, tell us about the problem you are experiencing or the feature you'd like to request.

#### Required information
In order to help us sort your issue, we ask that you provide the following information:
- [x] Mac model: MacPro 7,1 2019 Desktop Computer
- [x] macOS version: Ventura 13.5.2
- [x] Monitor setup: LG 5K Monitor with Thunderbolt Cable (Irrelevant to the issue)

If appropriate, please enable `Debug mode` and `Log to disk` in `Advanced` tab and replicate your bug, then attach the `AerialLog.txt` file (You can access this file through the Advanced tab).

#### Description of issue / Feature request

Aerial Companion Background App results in the failure of System Settings/Privacy/Full Disk Access not to load settings, affecting other Privacy settings resulting in many other Apps failing to load properly and displaying multiple windows on log-in when rebooting of the failure to access needed settings. FYI: It took me 3 solid days working ~18 hours a day on this to isolate this as no error messages were generated. This included several hours on the phone with Apple Senior Advisors, MacOS reinstall, Reseating all the boards and connector in the computer; reinstalling MacOS, and creating a new user folder from scratch.

**Comment by glouel** [comment]
Well, first, sorry you had an issue. 

Did you manage to replicate the issue afterwards ? Did removing the app from /Applications fixed it, or something else did ? Which version of companion was it (did it look like a menu ?) ? Did you install it in your menu bar or as a background updater only ?

This is the first time someone has reported anything like this, and the timing seems suspiciously close to this fairly similar issue : https://www.macrumors.com/2023/08/10/macos-ventura-13-5-location-services-bug/

Whatever they changed in 13.5 may have coincided with whatever you experienced, in general an app should not be able to break System Settings and Companion doesn't do much apart from setting a launch agent depending on your settings, but that not particularly unique either. 

Let me know if you have more information and I'll try to see if I can replicate the issue. Thanks!

**Comment by Gym-AW** [comment]
Thank you for the reply Yes I did replicate it. It was running as a menu item. The Ventura update also affected two other apps similarly later on as they were not installed when Aerial was running. I believe the version was 1.2.2. running on a MacPro Desktop computer 7.1 (2019). The other Apps that were the same issue are LuLu and Garmin Express. I “Fixed” Garmin Express by launching it on startup with a program called  Autostarter which allows a delay time be set during  login for an App and turning of its ability to run in the background which is part of Garmin Express. That gave Full Disk Access enough time to load so that Garmin Express would not interfere with it. I might try later the same ’trick’ with Aerial but right now I need to catch up on my work as it took me 3, 18 hour days to get my machine working again. I am still running Aerial as a screensaver and it is working fine. The issue was strictly Aerial Companion and I was able to replicate the issue. 


-----------------------
Regards,
James Weil
***@***.***

**Comment by glouel** [comment]
Thanks for your reply, this is super helpful. From this I think it's pretty clear to me it's the launch service thing is what tripped System Settings. 

Apple introduced a new API with the new System Settings, but it had many issues with apps using the old system (Companion being one of those apps) but most of the time benign (it showed the developer name instead of the name of the app for example).  

Things have been really weird since and many bugs popped up everywhere with many apps. 

Anyway, right now, I'm not reproducing this with the latest version of Aerial (it looks like this, which your version probably didn't) : 
![Capture d’écran 2023-09-11 à 20 34 20](https://github.com/JohnCoates/Aerial/assets/37544189/dd69ae7d-d17a-4328-a5d8-e628fe55edf7)

And this doesn't cause any issue in the Sonoma beta I'm running. So I'll try 1.2.2 on Sonoma, and then try 13.5.2 and let you know. 

Short term, if you want to use Companion, I would recommend launching it manually and not setting it to launch at startup, or using your Autostarter thing which likely bypass the whole login service system. I'll try and get back to you asap with more explanation on the whole thing. 

Again, sorry you got caught by this, I'll do my best to fix this in an upcoming update!

**Comment by Gym-AW** [comment]
Thank you for your prompt rely and clarification of the issue. It is far more information than I received from Apple Support. It goes along with their habit of employing users and developers as volunteer beta testers instead of spending the time and money to do proper SQA before release. Looking forward to the release. Right now since Aerial is functioning fine as my screensaver I really don’t need to use AC right now so it is not an urgent issue for the time being. 


-----------------------
Best Regards,
James Weil
***@***.***

**Comment by glouel** [comment]
Hey again @Gym-AW 

So I gave it a shot in a blank-ish virtual machine upgraded to 13.5.2, just installed Aerial Companion, and full disk access can be opened fine in that case, that is with Companion set to launch at startup : 

![Capture d’écran 2023-09-12 à 14 54 41](https://github.com/JohnCoates/Aerial/assets/37544189/c65ab51b-9b14-436c-b774-09bb1fafa528)

Did I roughly get your configuration right or did you set it to "Launch in the background (no menu)". I assumed you used this old version (1.2.2) of Companion (that was the download link for a couple of years).

![Capture d’écran 2023-09-12 à 15 05 17](https://github.com/JohnCoates/Aerial/assets/37544189/c4341f5d-e1d9-4183-95ab-c363173c462f)

One of the bug in System Settings that we see when using the old Launch Services is that it doesn't show the app name in the "open at startup" panel, but the developer name (which is weirding people out, rightly so !). 

I also tried with the really recently released 1.5.0b9 (I made it the download link for everyone last week) and it's the same (as far as I remember there was no change in those code paths so I don't think the version should matter much) : 

![Capture d’écran 2023-09-12 à 15 08 36](https://github.com/JohnCoates/Aerial/assets/37544189/4eac540b-8f54-4d81-bf97-3c749f9357e2)

Sonoma beta is also fine (for this precise issue at least).

I'm still fairly convinced based on your description that whatever you experienced is linked to the launch settings of apps, but maybe the bug gets triggered when you have a couple of them and not just one ? Or it's another app that broke it and you had to remove every app that ever installed a launch service to get it working again ? Or there's some extra factor at play. 

Let me know if I got it wrong with your settings or something, I'll keep an eye open in any case should someone report something similar.

**Comment by Gym-AW** [comment]
Thank you for the update. The reason it might have worked for you is that you are using an M1 CPU  as a virtual machine and I am using Intel on a MacPro 7.1. Often the issue manifests itself after a hardware shutdown and startup. I also chose Startup but you message gave me an idea which is to launch it with no menu or manually as I really rarely using it except for updates as most everything else can be done in screen saver.. 
--------------
Regards,
Jim Weil
***@***.***

**Comment by Gym-AW** [comment]
Just to be sure I installed it again and made sure it was V 1.2.2. The issue immediately presented itself and I tested it with all the workarounds that worked on some other apps and all of its launch options with no success. Unfortunately it was quite challenging to remove it as it had filled all over the place and it took me 2 hours to completely remove it and restore my machine to get Full Disk Access to properly work again. 

I simply no longer have the time or the patience for continued testing of any of the currently released versions. 
-----------------------
Regards,
James Weil
***@***.***

**Comment by glouel** [comment]
Did it present itself when you set it to run at startup/in the background ? Just launching the app should not do anything. 

In any case there's definitely something else going on to trigger whatever bug they have in System Settings. 

I'll be switching to the new API in Companion 2, I'll let you know should you want to try it out again, but there's little more I can do here but see if other people also have the issue.

**Comment by Gym-AW** [comment]
Yes. Before I reinstalled it my machine was stable. Since removing it that is no longer the case. You are correct in that something else is going on but Aerial seems to have triggered it. 

FYI Is their any way to install Aerial screensaver without the companion?
-----------------------
Regards,
James Weil
***@***.***

**Comment by glouel** [comment]
My question was, you can run companion without setting it to launch at startup, that should not cause any issue, so did you try that or did you set it to launch at startup?

There are multiple installation options yes, check this for all options : https://aerialscreensaver.github.io/installation.html

**Comment by Gym-AW** [comment]
Tried all of them with the same result. Full Disk Access did not populate resulting in my machine becoming unusable and sometimes locking up requiring a hard power off to restart it. 
-----------------------
Regards,
James Weil
***@***.***

**Comment by glouel** [comment]
Changed a lot of things since then, we no longer need full disk access nor background mode in V4, so whatever the issue was with 1.2.2, it should hopefully no longer occur.
